test/test_tensor_creation_ops.py

```diff
@@ -1938,6 +1938,11 @@ def test_constructor_device_legacy(self, device):
         self.assertRaises(RuntimeError, lambda: torch.Tensor(torch.Size([2, 3, 4]), device='cuda'))
         self.assertRaises(RuntimeError, lambda: torch.Tensor((2.0, 3.0), device='cuda'))
 
+        # Tensor constructor/new with Tensor argument shouldn't work with device specified
+        i = torch.tensor([1], device='cpu')
```

Comment:
For completeness we should probably add the use case in the issue (a CUDA tensor passed to `torch.Tensor(... device='cpu')`), too. It can be added in the section below which checks for CUDA availability

Reply:
sure, it can be in this section though because we don't actually check for CUDA here.